cassovary-core/src/main/scala/com/twitter/cassovary/util/collections/Order.scala

```diff
@@ -0,0 +1,65 @@
+package com.twitter.cassovary.util.collections
+
+/**
+ * Specialized typeclass for describing ordering between elements.
+ *
+ * Based on spire implementation.
+ */
+trait Order[@specialized(Int, Long) A] {
+  def eqv(x: A, y: A): Boolean = compare(x, y) == 0
+  def gt(x: A, y: A): Boolean = compare(x, y) > 0
+  def lt(x: A, y: A): Boolean = compare(x, y) < 0
+  def gteqv(x: A, y: A): Boolean = compare(x, y) >= 0
+  def lteqv(x: A, y: A): Boolean = compare(x, y) <= 0
+
+  def min(x: A, y: A): A = if (lt(x, y)) x else y
+  def max(x: A, y: A): A = if (gt(x, y)) x else y
+  def compare(x: A, y: A): Int
+
+  /**
+   * Defines an order on `B` by mapping `B` to `A` using `f` and using `A`s
+   * order to order `B`.
+   */
+  def on[@specialized(Int, Long) B](f: B => A): Order[B] = new MappedOrder(this)(f)
+
+  /**
+   * Defines an ordering on `A` where all arrows switch direction.
+   */
+  def reverse: Order[A] = new ReversedOrder(this)
+}
+
+private[collections] class MappedOrder[@specialized A, @specialized B](order: Order[B])(f: A => B) extends Order[A] {
```

Comment:
let's continue to specialize only on Int and Long for now, rather than all primitives